go-controller/pkg/ovn/pods.go

```diff
@@ -62,6 +62,21 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 			}
 		}
 	}
+
+	if config.HybridOverlay.Enabled {
+		// allocate all previously annoted hybridOverlay Distributed Router IP addresses. Allocation needs to happen here
+		// before a Pod Add event can be processed and be allocated a previously assigned hybridOverlay Distributed Router IP address.
+		// we do not support manually setting the hybrid overlay DRIP address
+		nodes, err := oc.watchFactory.GetNodes()
+		if err != nil {
+			return fmt.Errorf("failed to get nodes: %v", err)
+		}
+		for _, node := range nodes {
+			if err := oc.allocateHybridOverlayDRIP(node); err != nil {
+				return fmt.Errorf("cannot allocate hybridOverlay DRIP on node %s (%v)", node.Name, err)
+			}
+		}
+	}
```

Comment:
Any particular reason not to have this in the `syncNodes` function? Seems like the most appropriate placement since these are originally allocated from the node handler.

Reply:
there is a race, once syncPods finishes new pods can be schedualed on the node. It is possible that a new pod can be allocated the nodes hybrid overlay distributed router address. If we allocate the pods that exist during startup in syncPods() the race condition no longer exists.

Reply:
@jcaamano to echo Jacob; the HO DRIP comes from the pod subnet (used to be the .3 address, now it can be anything in the host's pod subnet) so if we did this in syncNodes() we'd have to iterate all pods (like syncPods()) to reserve their IPs, then try to allocate the HO DRIP. So either we do pod stuff in syncNodes(), or node stuff in syncPods()...

Reply:
@JacobTanenbaum `syncNodes` comes before `syncPods` because we do `WatchNodes` before `WatchPods` we we would still be allocating the HO DRIP before any pod event.

@dcbw that makes sense. I guess I was thinking that the HO DRIP would already be annotated on the node but I see how that might not be the case always.